docs/datamaps/index.md

```diff
@@ -0,0 +1,169 @@
+Alternatively, an empty `Optional` will lead to the value being completely removed.  
+
+An example of a remover that will remove a value with a specific key from a `Map`-based data map:
+```java
+public record MapRemover(String key) implements DataMapValueRemover<Item, Map<String, String>> {
+    public static final Codec<MapRemover> CODEC = Codec.STRING.xmap(MapRemover::new, MapRemover::key);
+    
+    @Override
+    public Optional<Map<String, String>> remove(Map<String, String> value, Registry<Item> registry, Either<TagKey<Item>, ResourceKey<Item>> source, Item object) {
+        final Map<String, String> newMap = new HashMap<>(value);
+        newMap.remove(key);
+        return Optional.of(newMap);
+    }
+}
+```
+
+With the remover above in mind, we're attaching maps of string to string to items. Take the following data map JSON file:
+```js
+{
+    "values": {
+        //highlight-start
+        "minecraft:carrot": {
+            "somekey1": "value1",
+            "somekey2": "value2"
+        }
+        //highlight-end
+    }
+}
+```
+That file will attach the map `[somekey1=value1, somekey2=value2]` to the `minecraft:carrot` item. Now, another datapack can come on top of it and remove just the value with the `somekey1` key, as such:
+```js
+{
+    "remove": {
+        // As the remover is decoded as a string, we can use a string as the value here. If it were decoded as an object, we would have needed to use an object.
+        //highlight-next-line
+        "minecraft:carrot": "somekey1"
+    }
+}
+```
+After the second datapack is read and applied, the new value attached to the `minecraft:carrot` item will be `[somekey2=value2]`.
+
+## Datagen
+Data maps can be generated through `DataMapProvider`.  
+You should extend that class, and then override the `generate` method to create your entries, similar to tag generation.
+
+Considering the drop healing example from the start, we could generate some values as follows:
+```java
+public class DropHealingGen extends DataMapProvider {
+
+    public DropHealingGen(PackOutput packOutput, CompletableFuture<HolderLookup.Provider> lookupProvider) {
+        super(packOutput, lookupProvider);
+    }
+
+    @Override
+    protected void gather() {
+        // In the examples below, we do not need to forcibly replace any value as that's the default behaviour since a merger isn't provided, so the third parameter can be false.
+
+        // If you were to provide a merger for your data map, then the third parameter will cause the old value to be overwritten if set to true, without invoking the merger
+        builder(DROP_HEALING)
+                // Always give entities that drop any item in the minecraft:fox_food tag 12 hearts
+                .add(ItemTags.FOX_FOOD, new DropHealing(12, 1f), false)
+                // Have a 10% chance of healing entities that drop an acacia boat by one point
+                .add(Items.ACACIA_BOAT.builtInRegistryHolder(), new DropHealing(1, 0.1f), false);
```

Comment:
How do soft dependencies work with data maps and how would they be datagenned assuming the mod is not present.

Reply:
There are overloads that accept raw resource location IDs if the registry entries aren't present.